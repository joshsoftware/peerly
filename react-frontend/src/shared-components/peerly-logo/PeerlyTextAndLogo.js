import React from "react";
import PropTypes from "prop-types";

import PeerlyLogo from "shared-components/peerly-logo/PeerlyLogo";
import PeerlyTextComponent from "shared-components/peerly-logo/PeerlyTextComponent";

const PeerlyTextAndLogo = ({ theme, fontSize }) => (
  <div className="text-center">
    <PeerlyLogo theme={theme} fontSize={fontSize} />
    <PeerlyTextComponent theme={theme} fontSize={fontSize} />
  </div>
);

PeerlyTextAndLogo.defaultProps = {
  theme: "dark",
  fontSize: "30px",
};

PeerlyTextAndLogo.propTypes = {
  theme: PropTypes.oneOf(["dark", "light"]),
  fontSize: PropTypes.string,
};

export default React.memo(PeerlyTextAndLogo);
