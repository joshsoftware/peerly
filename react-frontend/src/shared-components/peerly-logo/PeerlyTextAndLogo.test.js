import React from "react";
import { render } from "@testing-library/react";

import PeerlyTextAndLogo from "shared-components/peerly-logo/PeerlyTextAndLogo";

it("should equal to Peerly text", () => {
  const { asFragment } = render(
    <PeerlyTextAndLogo theme={{ variant: "dark" }} fontSize="48px" />
  );
  expect(asFragment()).toMatchSnapshot();
});
