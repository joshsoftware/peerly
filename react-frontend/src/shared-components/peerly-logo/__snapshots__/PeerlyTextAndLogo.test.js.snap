// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should equal to Peerly text 1`] = `
<DocumentFragment>
  .c1 {
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
  font-size: 48px;
  color: var(--white);
}

.c0 {
  text-align: center;
  margin-top: 15px;
  width: 50px;
  height: 50px;
  border: 3px solid var(--white);
  background-color: var(--atomic);
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
}

.c2 {
  color: var(--white);
  background-color: var(--atomic);
  font-size: 48px;
}

<div
    class="text-center"
  >
    <div
      class="c0 mx-auto"
      data-testid="Logo"
    >
      <div
        class="c1"
        data-testid="plusSignText"
        font-size="48px"
      >
        +
      </div>
    </div>
    <div
      class="c2"
      data-testid="PeerlyTextComponents"
      font-size="48px"
    >
      Peerly
    </div>
  </div>
</DocumentFragment>
`;
